mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,476 @@
+from mvpa2.algorithms.hyperalignment import Hyperalignment
+from mvpa2.measures.base import Measure
+from mvpa2.datasets import Dataset, vstack
+from mvpa2.mappers.staticprojection import StaticProjectionMapper
+from scipy.linalg import LinAlgError
+from mvpa2.measures.searchlight import sphere_searchlight
+from mvpa2.mappers.zscore import zscore
+from scipy.sparse import coo_matrix, dok_matrix, csc_matrix
+from mvpa2.featsel.helpers import FractionTailSelector, FixedNElementTailSelector
+
+if __debug__:
+    from mvpa2.base import debug
+
+def compute_feature_scores(datasets, exclude_from_model=None):
+    """
+    Takes a list of datasets and computes a magical feature
+    score for each feature in each dataset
+    :ref:`Haxby et al., Neuron (2011) <HGC+11>`
+    Parameters
+    ----------
+        datasets : list or tuple of datasets
+        
+        exclude_from_model: list of dataset indices that won't participate
+                    in voxel selection of others
+
+    Returns
+    -------
+        list : a list of feature scores; higher the score, better the feature
+
+    NOTE: This function assumes that the datasets are zscored
+    """
+    if exclude_from_model is None:
+        exclude_from_model=[]
+    feature_scores = [ np.zeros(sd.nfeatures) for sd in datasets ]
+    for i, sd in enumerate(datasets):
+        for j, sd2 in enumerate(datasets[i+1:]):
+            corr_temp= np.dot(sd.samples.T, sd2.samples)
+            if  j+i+1 not in exclude_from_model:                
+                feature_scores[i] = feature_scores[i] + np.max(corr_temp, axis = 1)
+            if i not in exclude_from_model:
+                feature_scores[j+i+1] = feature_scores[j+i+1] + np.max(corr_temp, axis = 0)
+    return feature_scores
+
+
+class HyperalignmentMeasure(Measure):
+    is_trained=True
+    def __init__(self, ndatasets, hyperalignment, scale=0.0, ref_ds=0,
+            featsel=1.0 ,full_matrix=True, use_same_features=False, 
+            exclude_from_model=None, dtype='float64', **kwargs):
+        Measure.__init__(self, **kwargs)
+        self.ndatasets = ndatasets
+        self.hyperalignment = hyperalignment
+        self.scale = scale
+        self.ref_ds = ref_ds
+        self.featsel = featsel
+        self.use_same_features = use_same_features
+        self.exclude_from_model = exclude_from_model
+        if self.exclude_from_model is None:
+            self.exclude_from_model = []
+        self.full_matrix = full_matrix
+        self.dtype = dtype
+
+    def _call(self, dataset):
+        ds = []
+        nsamples = dataset.nsamples/self.ndatasets
+        nfeatures = dataset.nfeatures
+        seed_index = np.where(dataset.fa.roi_seed)
+        if self.scale>0.0:
+            dist = np.sum(np.abs(dataset.fa.voxel_indices-dataset.fa.voxel_indices[seed_index]), axis=1)
+            dist = np.exp(-(self.scale*dist/np.float(max(dist)) )**2)
+            dataset.samples = dataset.samples*dist
```

Comment:
modifying input dataset inplace? evil!  make a shallow copy and then override its samples